staging_vespalib/src/vespa/vespalib/util/document_runnable.cpp

```diff
@@ -42,7 +46,7 @@ bool Runnable::onStop()
 bool Runnable::join() const
 {
     vespalib::MonitorGuard monitor(_stateLock);
-    if (_state == STARTING || _state == RUNNING) return false;
+    assert ((_state != STARTING) && (_state != RUNNING));
```

Comment:
Would it would make sense to convert the similar return branch in `stop()` to an assertion as well? These kinds of silent returns are perfect for hiding wrongful use of the class API.

Though I fear that we have usages where we conditionally start something on init, but always stop on destruction, meaning that an assertion might core the process during an exception unwind on startup...

Reply:
That might be a next step.